RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,213 @@
+---
+title: "Gene-level summarization with `tximeta` for non-mouse or non-human"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**CCDL 2021**
+
+In this notebook, we'll import the transcript expression quantification output from `salmon quant` using the [`tximeta`](https://bioconductor.org/packages/release/bioc/html/tximeta.html) package but for a species not human or mouse. 
+
+The steps for human and mouse are more straightforward and are covered in `02-gastric_cancer_tximeta-live.Rmd`. 
+The main difference for using `tximeta` for other species is that we need to set up the reference data in the form of a `linkedTxome` before running `tximeta()` function. 
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love _et al_.
+
+## Libraries and functions
+
+```{r library}
+# Load magrittr for the pipe
+library(magrittr)
+
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+In this instance, we will need to provide `tximeta` with the salmon index for the species we are using. 
+We have prepared this for zebrafish along with other species which you can find more information about [on GitHub](https://github.com/AlexsLemonade/training-txome-prep).
+
+```{r directories}
+# Set up file path to the salmon index for the species we are using
+index_dir <- file.path("index", "Danio_rerio", "short_index")
+```
+
+These directories and files are for pointing to the `quant.sf` files. 
+
+```{r}
+# directory where the data are located
+data_dir <- file.path("data", "zebrafish-cortisol")
+
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "zebrafish-cortisol_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names}
+# Name the output zebrafish-cortisol_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "zebrafish-cortisol_tximeta.RDS")
+```
+
+## Set up reference data for a non-human, non-mouse species 
+
+Here we will set this up for zebrafish, _Danio rerio_.
```

Comment:
We should add some context here about the ensembl versions. Why are we using release 98 here? Because that was the version used when we made the salmon index. If they make their own index, they should know that this would be different.

We might actually want to provide a bit of info here about building an index; while we provide some, we might not have what somebody wants. For the ones that we do provide, we might think about having a table with links to the relevant metadata, rather than making people search for it.

A question I have: Is it possible to find the version info from the index itself, and if so, how? If it is possible, this is information we might want to provide.

Reply:
There does not appear to be an easy way to extract the original genome info from the salmon index. So I just tried to descibe this issue and link to our github `training-txome-prep` repo where relevant.